File: AI-agent-backend-fastapi/app/schemas/prompt_template.py
# Copyright (c) 2025 左岚. All rights reserved.
"""提示词模板Schema"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class PromptTemplateBase(BaseModel):
    """提示词模板基础模型"""
    name: str = Field(..., description="模板名称")
    template_type: str = Field("testcase_generation", description="模板类型")
    test_type: Optional[str] = Field(None, description="测试类型: API/Web/App/通用")
    content: str = Field(..., description="提示词内容")
    variables: Optional[str] = Field(None, description="变量说明(JSON格式)")
    description: Optional[str] = Field(None, description="模板描述")
    is_active: bool = Field(True, description="是否启用")


class PromptTemplateCreate(PromptTemplateBase):
    """创建提示词模板"""
    pass


class PromptTemplateUpdate(BaseModel):
    """更新提示词模板"""
    name: Optional[str] = None
    template_type: Optional[str] = None
    test_type: Optional[str] = None
    content: Optional[str] = None
    variables: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None


class PromptTemplateResponse(PromptTemplateBase):
    """提示词模板响应"""
    template_id: int
    is_default: bool
    created_by: Optional[int] = None
    create_time: datetime
    modify_time: Optional[datetime] = None

    class Config:
        from_attributes = True

File: AI-agent-frontend-vue3/src/api/prompt-template.ts
// Copyright (c) 2025 左岚. All rights reserved.
/**
 * 提示词模板管理 API
 */
import request from './request'

// ==================== 类型定义 ====================

export interface PromptTemplate {
    template_id: number
    name: string
    template_type: string
    test_type?: string
    content: string
    variables?: string
    description?: string
    is_default: boolean
    is_active: boolean
    created_by?: number
    create_time: string
    modify_time?: string
}

export interface PromptTemplateCreate {
    name: string
    template_type?: string
    test_type?: string
    content: string
    variables?: string
    description?: string
    is_active?: boolean
}

export interface PromptTemplateUpdate {
    name?: string
    template_type?: string
    test_type?: string
    content?: string
    variables?: string
    description?: string
    is_active?: boolean
}

// ==================== API 接口 ====================

/**
 * 创建提示词模板
 */
export function createPromptTemplateAPI(data: PromptTemplateCreate) {
    return request<PromptTemplate>({
        url: '/api/v1/prompt-templates',
        method: 'post',
        data
    })
}

/**
 * 获取提示词模板列表
 */
export function getPromptTemplatesAPI(params?: {
    template_type?: string
    test_type?: string
    is_active?: boolean
}) {
    return request<PromptTemplate[]>({
        url: '/api/v1/prompt-templates',
        method: 'get',
        params
    })
}

/**
 * 获取默认提示词模板
 */
export function getDefaultPromptTemplateAPI(params?: {
    template_type?: string
    test_type?: string
}) {
    return request<PromptTemplate>({
        url: '/api/v1/prompt-templates/default',
        method: 'get',
        params
    })
}

/**
 * 获取提示词模板详情
 */
export function getPromptTemplateAPI(templateId: number) {
    return request<PromptTemplate>({
        url: `/api/v1/prompt-templates/${templateId}`,
        method: 'get'
    })
}

/**
 * 更新提示词模板
 */
export function updatePromptTemplateAPI(templateId: number, data: PromptTemplateUpdate) {
    return request<PromptTemplate>({
        url: `/api/v1/prompt-templates/${templateId}`,
        method: 'put',
        data
    })
}

/**
 * 删除提示词模板
 */
export function deletePromptTemplateAPI(templateId: number) {
    return request({
        url: `/api/v1/prompt-templates/${templateId}`,
        method: 'delete'
    })
}

/**
 * 设置为默认模板
 */
export function setDefaultPromptTemplateAPI(templateId: number) {
    return request<PromptTemplate>({
        url: `/api/v1/prompt-templates/${templateId}/set-default`,
        method: 'post'
    })
}

